NPE in query.impl.TableClientImpl happens on query error

```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "tech.ydb.proto.query.YdbQuery$TransactionMeta.getId()" because "meta" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:636) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194) ~[?:?]
...
Caused by: java.lang.NullPointerException: Cannot invoke "tech.ydb.proto.query.YdbQuery$TransactionMeta.getId()" because "meta" is null
	at tech.ydb.query.impl.TableClientImpl$TableSession$1.handleTxMeta(TableClientImpl.java:163) ~[ydb-sdk-query-2.3.4.jar:?]
	at tech.ydb.query.impl.SessionImpl$StreamImpl.lambda$execute$0(SessionImpl.java:273) ~[ydb-sdk-query-2.3.4.jar:?]
	at tech.ydb.core.impl.call.ReadStreamCall.onMessage(ReadStreamCall.java:96) ~[ydb-sdk-core-2.3.4.jar:?]
	... 7 more
```

This happens, for example, when SELECT completes with exception in the interactive transaction.

--- a/query/src/main/java/tech/ydb/query/impl/SessionImpl.java
+++ b/query/src/main/java/tech/ydb/query/impl/SessionImpl.java
@@ -210,8 +210,7 @@ abstract class SessionImpl implements QuerySession {
         YdbQuery.TransactionControl tc = TxControl.txModeCtrl(tx, true);
         return new StreamImpl(createGrpcStream(query, tc, prms, settings)) {
             @Override
-            void handleTxMeta(YdbQuery.TransactionMeta meta) {
-                String txID = meta == null ? null : meta.getId();
+            void handleTxMeta(String txID) {
                 if (txID != null && !txID.isEmpty()) {
                     logger.warn("{} got unexpected transaction id {}", SessionImpl.this, txID);
                 }
@@ -253,7 +252,7 @@ abstract class SessionImpl implements QuerySession {
             this.grpcStream = grpcStream;
         }
 
-        abstract void handleTxMeta(YdbQuery.TransactionMeta meta);
+        abstract void handleTxMeta(String txId);
         void handleCompletion(Status status, Throwable th) { }
 
         @Override
@@ -276,7 +275,7 @@ abstract class SessionImpl implements QuerySession {
                 }
 
                 if (msg.hasTxMeta()) {
-                    handleTxMeta(msg.getTxMeta());
+                    handleTxMeta(msg.getTxMeta().getId());
                 }
                 if (issues.length > 0) {
                     if (handler != null) {
@@ -352,8 +351,8 @@ abstract class SessionImpl implements QuerySession {
 
             return new StreamImpl(createGrpcStream(query, tc, prms, settings)) {
                 @Override
-                void handleTxMeta(YdbQuery.TransactionMeta meta) {
-                    String newId = meta == null || meta.getId() == null || meta.getId().isEmpty() ? null : meta.getId();
+                void handleTxMeta(String txID) {
+                    String newId = txID == null || txID.isEmpty() ? null : txID;
                     if (!txId.compareAndSet(currentId, newId)) {
                         logger.warn("{} lost transaction meta id {}", SessionImpl.this, newId);
                     }

--- a/query/src/main/java/tech/ydb/query/impl/TableClientImpl.java
+++ b/query/src/main/java/tech/ydb/query/impl/TableClientImpl.java
@@ -156,17 +156,17 @@ public class TableClientImpl implements TableClient {
                     .withRequestTimeout(settings.getTimeoutDuration())
                     .build();
 
-            final AtomicReference<String> txID = new AtomicReference<>("");
+            final AtomicReference<String> txRef = new AtomicReference<>("");
             QueryStream stream = querySession.new StreamImpl(querySession.createGrpcStream(query, tc, prms, qs)) {
                 @Override
-                void handleTxMeta(YdbQuery.TransactionMeta meta) {
-                    txID.set(meta.getId());
+                void handleTxMeta(String txID) {
+                    txRef.set(txID);
                 }
             };
 
             return QueryReader.readFrom(stream)
                     .thenApply(r -> r.map(
-                            reader -> new ProxedDataQueryResult(txID.get(), reader)
+                            reader -> new ProxedDataQueryResult(txRef.get(), reader)
                     ));
         }
 

